apps/desktop-wallet/src/components/IOList.tsx

```diff
@@ -56,14 +54,13 @@ const IOList = ({
   const { t } = useTranslation()
   const internalAddressHashes = useAppSelector(selectAddressIds) as AddressHash[]
   const explorerUrl = useAppSelector((state) => state.network.settings.explorerUrl)
-
-  const [selectedAddressHash, setSelectedAddressHash] = useState<AddressHash>()
+  const dispatch = useAppDispatch()
```

Comment:
It's not possible to separate the modal name from the modal props because then we'll be able to pass a modal name that does not correspond to the right modal props. They need to be kept in the same object. In your example they are separated into 2 vars in the arguments of `openModal`. I'll look into it.

Reply:
Honestly I like `useModal` quite a bit because you can clearly see right at the top of the component that it can trigger the opening of a modal. A bit like when you see `useTranslation`, `useTheme`, it's easy to spot the side effects and dependencies this way.

Reply:
I hear you, and I see the merit of your proposal, but I'd like to stick to my proposal and arguments, I am not convinced that it's worth creating `useModal`. I don't see enough benefits. Uniswap also doesn't have it. I'd like to keep it simple with just a `dispatch`.

Reply:
I experimented a bit more with your proposal, specifically I tried creating `useModalOpen`. I discovered another reason why I don't like it. By using the `dispatch` method we can define our `open[...]Modal` functions **after** the `return null` line in many components, saving us some more memory from declaring unnecessary functions. The `useModal` being a hook needs to always be declared above conditions.

Worth mentioning that all the above are nitpicking IMO :) We are talking about tiny optimizations and tiny improvements.

Reply:
Works for me. I've kept the `dispatch` in MW too ;)